### ISSUE TITLE
Rule 951240 triggering false positive in Wordpress widgets.php

### ISSUE THREAD

**Issue by Fregf** [coreruleset/coreruleset#1875]
### Description

When opening /wp-admin/widgets.php with both Wordpress 5.4 and Wordpress 5.5 this triggers a false positivie in CRS, resulting in a 403 Forbidden error. I attach the contents of mod_security audit log. Here is Apache's error log:
```
[Sat Sep 05 20:59:44.774814 2020] [:error] [pid 22811:tid 140300701771520] [client xxx.xxx.xxx.xxx:39968] [client xxx.xxx.xxx.xxx] ModSecurity: Rule 7f9a9941b6e8 [id "-"][file "/usr/share/modsecurity-crs/rules/RESPONSE-951-DATA-LEAKAGES-SQL.conf"][line "92"] - Execution error - PCRE limits exceeded (-8): (null). [hostname "example.org"] [uri "/wp-admin/widgets.php"] [unique_id "X1PgIEwWLLinfZz@i-mbGgAAAqo"], referer: https://example.org/wp-admin/index.php
[Sat Sep 05 20:59:44.962742 2020] [:error] [pid 22811:tid 140300701771520] [client xxx.xxx.xxx.xxx:39968] [client xxx.xxx.xxx.xxx] ModSecurity: Warning. Pattern match "(?i:PostgreSQL query failed:|pg_query\\\\(\\\\) \\\\[:|pg_exec\\\\(\\\\) \\\\[:|PostgreSQL.*ERROR|Warning.*pg_.*|valid PostgreSQL result|Npgsql\\\\.|PG::[a-zA-Z]*Error|Supplied argument is not a valid PostgreSQL .*? resource|Unable to connect to PostgreSQL server)" at RESPONSE_BODY. [file "/usr/share/modsecurity-crs/rules/RESPONSE-951-DATA-LEAKAGES-SQL.conf"] [line "390"] [id "951240"] [msg "postgres SQL Information Leakage"] [data "Matched Data: warning\\x22>Add-Ons</span></a></li><li><a href='admin.php?page=wps_donate_page'><span class=\\x22wps-text-success\\x22>Donate</span></a></li></ul></li>\\x0a\\x09<li class=\\x22wp-has-submenu wp-not-current-submenu menu-top toplevel_page_mlang menu-top-last\\x22 id=\\x22toplevel_page_mlang\\x22><a href='admin.php?page=mlang' class=\\x22wp-has-submenu wp-not-current-submenu menu-top toplevel_page_mlang menu-top-last\\x22 aria-haspopup=\\x22true\\x22><div class=\\x22wp-menu-arrow\\x22><div></div></div><div ..."] [severity "CRITICAL"] [ver "OWASP_CRS/3.3.0"] [tag "applicati [hostname "example.org"] [uri "/wp-admin/widgets.php"] [unique_id "X1PgIEwWLLinfZz@i-mbGgAAAqo"], referer: https://example.org/wp-admin/index.php
[Sat Sep 05 20:59:44.970342 2020] [:error] [pid 22811:tid 140300701771520] [client xxx.xxx.xxx.xxx:39968] [client xxx.xxx.xxx.xxx] ModSecurity: Rule 7f9a99342640 [id "-"][file "/usr/share/modsecurity-crs/rules/RESPONSE-951-DATA-LEAKAGES-SQL.conf"][line "420"] - Execution error - PCRE limits exceeded (-8): (null). [hostname "example.org"] [uri "/wp-admin/widgets.php"] [unique_id "X1PgIEwWLLinfZz@i-mbGgAAAqo"], referer: https://example.org/wp-admin/index.php
[Sat Sep 05 20:59:45.026576 2020] [:error] [pid 22811:tid 140300701771520] [client xxx.xxx.xxx.xxx:39968] [client xxx.xxx.xxx.xxx] ModSecurity: Access denied with code 403 (phase 4). Operator GE matched 4 at TX:outbound_anomaly_score. [file "/usr/share/modsecurity-crs/rules/RESPONSE-959-BLOCKING-EVALUATION.conf"] [line "76"] [id "959100"] [msg "Outbound Anomaly Score Exceeded (Total Score: 5)"] [ver "OWASP_CRS/3.3.0"] [tag "anomaly-evaluation"] [hostname "example.org"] [uri "/wp-admin/widgets.php"] [unique_id "X1PgIEwWLLinfZz@i-mbGgAAAqo"], referer: https://example.org/wp-admin/index.php
[Sat Sep 05 20:59:45.027387 2020] [:error] [pid 22811:tid 140300701771520] [client xxx.xxx.xxx.xxx:39968] [client xxx.xxx.xxx.xxx] ModSecurity: Warning. Operator GE matched 4 at TX:outbound_anomaly_score. [file "/usr/share/modsecurity-crs/rules/RESPONSE-980-CORRELATION.conf"] [line "102"] [id "980140"] [msg "Outbound Anomaly Score Exceeded (score 5): individual paranoia level scores: 5, 0, 0, 0"] [ver "OWASP_CRS/3.3.0"] [tag "event-correlation"] [hostname "example.org"] [uri "/wp-admin/widgets.php"] [unique_id "X1PgIEwWLLinfZz@i-mbGgAAAqo"], referer: https://example.org/wp-admin/index.php
```

CRS 3.3.0, libapache2-mod-security2 2.9.3-1, Debian 10 Buster

I disabled this rule in wordpress exclusion file for widgets.php, because wordpress works only with mysql anyway.

**Comment by Fregf** [comment]
[audit.log](https://github.com/coreruleset/coreruleset/files/5178508/audit.log)

**Comment by azurit** [comment]
Can you tell me ID of the rule you disabled?

**Comment by dune73** [comment]
Unfortunately, original submitter never explained his submission. It is not quite clear. But here is what I got:

A request to `/wp-admin/widgets.php` triggers an FP `951240 postgres SQL Information Leakage` and the original submitter had to write a rule exclusion for this situation. We should include such a rule exclusion in our wordpress rule exclusion package at `REQUEST-903.9002-WORDPRESS-EXCLUSION-RULES.conf`.

**Comment by dune73** [comment]
We are not currently doing any response rule exclusions. I also think the SQL error message might in fact be valid and point to a misconfiguration and then it would not be a FP.

Also, I can not seem to trigger the rule with the http response found in the audit log (thank you for submitting).

This is the response body in question:

[test-1875.txt](https://github.com/coreruleset/coreruleset/files/5830463/test-1875.txt)

(Uploading as .txt, since github won't accept .html).

In absence of a way to reproduce the issue, I am closing this issue. Feel free to open it again if you have more information.
